mixed-content/generic/worker.js

```diff
@@ -0,0 +1,7 @@
+try {
+  // Works on same origin.
+  postMessage('done');
+} catch(err) {
+  // Works cross-origin.
+  postMessage('done', '*');
```

Comment:
Done!

So to conclude, the issue was:

I'm mapping the content type (mime type) to what expect.py will end up
producing, since application/javascript mapped to worker.js, it was being
executed in the top-level context and postMessage was not happy about it.

Therefore even an empty catch block would have worked fine.

So for script-tag I'm now just using text/plain since we catch onload and
onerror events anyway. In any other case I would have to add another
parameter other than contentType.



On Wed, Jul 1, 2015 at 12:08 PM, Mike West <notifications@github.com> wrote:

> In mixed-content/generic/worker.js
> <https://github.com/w3c/web-platform-tests/pull/1889#discussion_r33665249>
> :
>
> > @@ -0,0 +1,7 @@
> > +try {
> > +  // Works on same origin.
> > +  postMessage('done');
> > +} catch(err) {
> > +  // Works cross-origin.
> > +  postMessage('done', '*');
>
>
>    1.
>
>    I don't believe it's possible to load an HTTPS worker on an HTTP page
>    (see step #5 <https://github.com/w3c/web-platform-tests/pull/5> of
>    https://html.spec.whatwg.org/multipage/workers.html#dom-worker (note
>    that Firefox allows data: Workers, but it also treats data: as the
>    same-origin as its embedder; Chrome does not, and does not)).
>    2.
>
>    Regardless, a worker's postMessage always talks to its embedder; it
>    doesn't have an origin flag (the second argument is a set of objects to
>    transfer from one context to the other: see
>    https://html.spec.whatwg.org/multipage/workers.html#dedicated-workers-and-the-dedicatedworkerglobalscope-interface:dom-dedicatedworkerglobalscope-postmessage
>    ).
>
> You can simplify this file to postMessage('done');.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/w3c/web-platform-tests/pull/1889/files#r33665249>.
>



-- 

*Kristijan Burnik*

Software Engineering Intern

burnik@google.com

Google Germany GmbH

Dienerstraße 12

80331 München

Geschäftsführer: Graham Law, Christine Elizabeth Flores

Registergericht und -nummer: Hamburg, HRB 86891

Sitz der Gesellschaft: Hamburg